tests/integration_tests/test_hyperopt.py

```diff
@@ -126,4 +128,69 @@ def test_hyperopt_executor_with_metric(csv_filename):
                            validate_output_feature=True,
                            validation_metric=ACCURACY)
 
-# todo future: add test for hyperopt.run.hyperopt function
+def test_hyperopt_run_hyperopt(csv_filename):
+    input_features = [
+        text_feature(name="utterance", cell_type="lstm", reduce_output="sum"),
+        category_feature(vocab_size=2, reduce_input="sum")]
+
+    output_features = [category_feature(vocab_size=2, reduce_input="sum")]
+
+    rel_path = generate_data(input_features, output_features, csv_filename)
+
+    model_definition = {
+        "input_features": input_features,
+        "output_features": output_features,
+        "combiner": {"type": "concat", "num_fc_layers": 2},
+        "training": {"epochs": 2, "learning_rate": 0.001}
+    }
+
+    output_feature_name = output_features[0]['name']
+
+    hyperopt_configs = {
+        "parameters": {
+            "training.learning_rate": {
+                "type": "float",
+                "low": 0.0001,
+                "high": 0.01,
+                "space": "log",
+                "steps": 3,
+            },
+            output_feature_name + ".fc_size": {
+                "type": "int",
+                "low": 32,
+                "high": 256,
+                "steps": 5
+            },
+            output_feature_name + ".num_fc_layers": {
+                'type': 'int',
+                'low': 1,
+                'high': 5,
+                'space': 'linear',
+                'steps': 4
+            }
+        },
+        "goal": "minimize",
+        'output_feature': output_feature_name,
+        'validation_metrics': 'loss',
+        'executor': {'type': 'serial'},
+        'sampler': {'type': 'random', 'num_samples': 5}
```

Comment:
To keep this fast, 2 samples is sufficient, we are already testing more samples with the other hyperopt tests

Reply:
done.